cogs/cipher.py

```diff
@@ -0,0 +1,111 @@
+                      usage_information=[
+                          usage_info("", "If a shift value is not found at the start "
+                                         "of the message it would try and find it at the end")
+                      ],
+                      examples=[argument("Hello", "Returns Khoor"), argument("10 Hello", "Returns Rovvy")])
+    async def shift(self, ctx, shift: typing.Optional[int], *, message):
+        if not shift:
+            M = message.split()
+            try:
+                shift = int(M[-1])
+                message = " ".join(M[:-1])
+            except ValueError:
+                shift = 3
+
+        if not message:
+            return await ctx.send(f"I need some text to shift by {shift} places!")
+
+        # If the shift is higher than 25 then this would convert the shift to a value that is lower than 25
+        shift %= 26
+
+        shifted_characters = list(map(lambda m: m[shift:] + m[:shift], self.character_set))
+        translation_rules = str.maketrans("".join(self.character_set), "".join(shifted_characters))
+        await ctx.send(message.translate(translation_rules))
+
+    @commands.command(cls=Command,
+                      description="Flips the alphabet, turns A to Z, B to Y...",
+                      syntax=[argument(True, "Message")],
+                      examples=[example("Hello", "Returns \"Svool\"")])
+    async def atbash(self, ctx, *, message):
+        # Reverses the character set
+        reversed_characters = list(map(lambda m: m[::-1], self.character_set))
+        translation_rules = str.maketrans("".join(self.character_set), "".join(reversed_characters))
+        await ctx.send(message.translate(translation_rules))
+
+    @commands.command(cls=Command,
+                      description="Encode a message in base64!",
+                      syntax=[argument(True, "Message")],
+                      usage_information=[
+                          usage_info("", "Conversion is done via encoding the UTF-8 "
+                                         "codepoint to base64 for each letter in the message")
+                      ],
+                      examples=[example("Hello", "Return SGVsbG8=")])
+    async def encode64(self, ctx, *, message):
+        b64msg = base64.b64encode(message.encode("utf-8")).decode("utf-8")
+        await ctx.send(b64msg)
+
+    @commands.command(cls=Command,
+                      description="Decodes a message in base64",
+                      syntax=[argument(True, "Message")],
+                      usage_information=[
+                          usage_info("", "Conversion is done via encoding the UTF-8 "
+                                         "codepoint to base64 for each letter in the message")
+                      ],
+                      examples=[example("SGVsbG8=", "Returns \"Hello\"")])
+    async def decode64(self, ctx, *, message):
+        """Decodes a message in base64
```

Comment:
D205: 1 blank line requierd between summary line and description (found 0)
D400: First line should end with a period (not '4')
D401: First line should be in imperative mood (perhaps 'Decode', not 'Decodes')